flapjack/ext.py

```diff
@@ -200,3 +200,16 @@ def flatpak_builder(*args, check=None, distcheck=False):
 
     with _BranchAllModules():
         return subprocess.call(cmdline, cwd=config.workdir())
+
+
+def get_ps1():
+    """Try to obtain the value of the $PS1 variable from the parent shell.
+    Fallback to a simple prompt if not found."""
+
+    shell = os.environ.get('SHELL')
```

Comment:
Have you tried `os.environ.get('PS1')`?

Reply:
@ptomato yep, that was my initial thought as well. The environment variables are not passed to Python unless explicitely exported. For example in my system:

```
[manuq@localhost flapjack]$ python3 -c 'import os; print(os.environ.get("PS1"))'
None
[manuq@localhost flapjack]$ PS1=$PS1 python3 -c 'import os; print(os.environ.get("PS1"))'
[\u@\h \W]\$ 
```

Reply:
How bizarre! It works for me:
```
(T20603) ~/checkout/hatch-previewer$ python3 -c 'import os; print(os.environ.get("PS1"))'
\[\e[33;41;1m\]`if [ $? -ne 0 ]; then echo "⚔ "; fi`\[\e[0m\]\[\e[32m\]\[\e[0m\]$(__git_ps1 "(\[$YELLOW\]%s\[$NORMAL\]) ")\[\e[35m\]`if [[ $(pwd|sed -e "s,^$HOME,~,"|wc -c|tr -d " ") > 32 ]]; then echo "\W"; else echo "\w"; fi`$ \[\e[0m\]
```

Reply:
Whoa, I guess I can try to get "PS1" first, then try the method I implemented, and finally fallback to a simple "$ " prompt. Or just try PS1, then fallback to "$ ", to avoid adding hackish code. What do you think?

Reply:
Either one is fine with me. I would really like to know the reason for this difference though!

Reply:
@ptomato I will add the workaround because I asked on IRC and others got None too for the test above. Maybe you have `PYTHONSTARTUP` set in your environment? You could check by running the test above with -E (avoid modifying the behaviour of the interpreter).  Or even with -I (isolated mode).